src/xdsl/ir.py

```diff
@@ -232,6 +236,12 @@ class Operation:
     parent: Optional[Block] = field(default=None)
     """The block containing this operation."""
 
+    irdl_operand_defs: List[Tuple[str, OperandDef]] = []
```

Comment:
Here, you should write `= field(default_factory=list)`.
This is because this sets a mutable variable as default